# Module 5

### Notes

## Binomial

### binom.pmf(x,n,p) (PMF), binom.cdf(x,n,p) (CDF)

In [1]:
from scipy.stats import binom

# This example uses n=10, p=0.4, and evaluates at x=3

n = 10
p = 0.4
x = 3

pmf_value = binom.pmf(x, n, p)
cdf_value = binom.cdf(x, n, p)

print(f"PMF (x={x}) = {pmf_value:.4f}")
print(f"CDF (x<={x}) = {cdf_value:.4f}")

PMF (x=3) = 0.2150
CDF (x<=3) = 0.3823


## Bernoulli

In [24]:
from scipy.stats import binom

p = 0.3   # probability of success
n = 1     # one trial

# x = 0 (failure)
pmf_0 = binom.pmf(0, n, p)
print(f"pmf(x=0) = {pmf_0:.4f}")

# x = 1 (success)
pmf_1 = binom.pmf(1, n, p)
print(f"pmf(x=1) = {pmf_1:.4f}")

pmf(x=0) = 0.7000
pmf(x=1) = 0.3000


## Hypergeometric

Scenario:
A box has N = 20 items, R = 7 of them are defective.
You randomly inspect n = 5 items.
What is the probability that x = 2 of them are defective?

In [2]:
from scipy.stats import hypergeom

N = 20   # population size
R = 7    # number of success states in population
n = 5    # number of draws
x = 2    # observed successes

pmf_value = hypergeom.pmf(x, N, R, n)
cdf_value = hypergeom.cdf(x, N, R, n)

print(f"PMF (x={x}) = {pmf_value:.4f}")
print(f"CDF (x<={x}) = {cdf_value:.4f}")

PMF (x=2) = 0.3874
CDF (x<=2) = 0.7932


 Interpretation
- 	PMF = 0.3241 → There is about a 32.41% chance of finding exactly 2 defective items in your 5‑item sample.
- 	CDF = 0.7905 → There is about a 79.05% chance of finding 0, 1, or 2 defective items

Full Distribution Table

In [3]:
import numpy as np

xs = np.arange(0, n+1)
pmf_all = hypergeom.pmf(xs, N, R, n)
cdf_all = hypergeom.cdf(xs, N, R, n)

for xi, pm, cd in zip(xs, pmf_all, cdf_all):
    print(f"x={xi:2d}  PMF={pm:.4f}  CDF={cd:.4f}")

x= 0  PMF=0.0830  CDF=0.0830
x= 1  PMF=0.3228  CDF=0.4058
x= 2  PMF=0.3874  CDF=0.7932
x= 3  PMF=0.1761  CDF=0.9693
x= 4  PMF=0.0293  CDF=0.9986
x= 5  PMF=0.0014  CDF=1.0000


## Poisson
Parameters
- 	x — number of observed events
- 	mu (λ) — average event rate (mean and variance of the distribution)

Scenario:
A call center receives an average of λ = 4 calls per hour.
What is the probability of receiving exactly 2 calls, and what is the probability of receiving 2 or fewer calls?

In [4]:
from scipy.stats import poisson

mu = 4   # average rate
x = 2    # observed events

pmf_value = poisson.pmf(x, mu)
cdf_value = poisson.cdf(x, mu)

print(f"PMF (x={x}) = {pmf_value:.4f}")
print(f"CDF (x<={x}) = {cdf_value:.4f}")

PMF (x=2) = 0.1465
CDF (x<=2) = 0.2381


Interpretation
- 	PMF = 0.1465 → About a 14.65% chance of receiving exactly 2 calls in an hour.
- 	CDF = 0.2381 → About a 23.81% chance of receiving 0, 1, or 2 calls.

Full distribution table

In [5]:
import numpy as np

xs = np.arange(0, 10)
pmf_all = poisson.pmf(xs, mu)
cdf_all = poisson.cdf(xs, mu)

for xi, pm, cd in zip(xs, pmf_all, cdf_all):
    print(f"x={xi:2d}  PMF={pm:.4f}  CDF={cd:.4f}")

x= 0  PMF=0.0183  CDF=0.0183
x= 1  PMF=0.0733  CDF=0.0916
x= 2  PMF=0.1465  CDF=0.2381
x= 3  PMF=0.1954  CDF=0.4335
x= 4  PMF=0.1954  CDF=0.6288
x= 5  PMF=0.1563  CDF=0.7851
x= 6  PMF=0.1042  CDF=0.8893
x= 7  PMF=0.0595  CDF=0.9489
x= 8  PMF=0.0298  CDF=0.9786
x= 9  PMF=0.0132  CDF=0.9919


## Geometric
A geometric example in SciPy uses the idea that the distribution models the trial number on which the first success occurs.

Scenario:
A process has a success probability p = 0.3 on each independent trial.
What is the probability that the first success occurs on trial x = 4, and what is the probability that it occurs on or before trial 4?

In [6]:
from scipy.stats import geom

p = 0.3
x = 4

pmf_value = geom.pmf(x, p)
cdf_value = geom.cdf(x, p)

print(f"PMF (x={x}) = {pmf_value:.4f}")
print(f"CDF (x<={x}) = {cdf_value:.4f}")

PMF (x=4) = 0.1029
CDF (x<=4) = 0.7599


 Interpretation
- 	PMF gives the probability that the first success happens exactly on trial 4.
- 	CDF gives the probability that the first success happens on trial 1, 2, 3, or 4.


Full distribution table